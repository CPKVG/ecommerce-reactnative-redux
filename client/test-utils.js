// import { configureStore } from '@reduxjs/toolkit';
// import React from 'react';
// import { render as rtlRender } from '@testing-library/react-native'
// import { Provider } from 'react-redux';
// import { PersistGate } from 'redux-persist/integration/react';
// import { persistor } from './redux/createStore';
// import { rootReducer } from './redux/rootReducer';

// //custom render

// function render(
//     ui ,{
//         preloadedState,
//         store = configureStore({ reducer: rootReducer, preloadedState}),
//         ...renderOptions
//     } = {}
// ){
//   function Wrapper( {childern} ){
//       return <Provider store = {store}>
//           <PersistGate persistor={persistor}>
//             {childern}
//           </PersistGate>
//       </Provider>
//   }  
//   return rtlRender(ui, {wrapper: Wrapper, ...renderOptions})
// }

// //re-export everything
// export * from '@testing-library/react-native'
// //override render method
// export { render }


import React from "react"
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
// import configureStore from 'redux-mock-store';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { rootReducer } from "./redux/rootReducer";

const INITIAL_STATE = { ...rootReducer }; // Or use value from root reducer
const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)
// const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
  render(ui, {
    
    wrapper: ({ children }) => <
      Provider store={store}>
          {children}
      </Provider>,
    ...options
  });

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender as render, store };