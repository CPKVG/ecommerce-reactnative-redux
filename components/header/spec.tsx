import { NavigationContainer } from '@react-navigation/native'
import React from 'react'
import Header from '.'
import { render,cleanup, fireEvent } from '../../test-utils'
import renderer from "react-test-renderer"
import { Text, View } from 'react-native'

afterEach(cleanup)
jest.useFakeTimers()


describe("header component",()=>{
    let title = "test Title"
    const component = (
        <NavigationContainer>
            <Header title={title} />
        </NavigationContainer>
    )

    it("should render the title text",()=>{
        const { findByText, getByText } = render(component)
        const button = getByText('test Title')
                
        fireEvent.press(button)
        expect(button).toBeTruthy
        //unable to detech any
        // const newScreen = findByText('')
        // expect(newScreen).toBeTruthy    
    })

    it("renders the Header component",() => {
        const tree = renderer.create(
            <View>
                <Text>
                    {title}
                </Text>
          </View>
        ).toJSON();
        expect(tree).toMatchSnapshot();
        })
    
})