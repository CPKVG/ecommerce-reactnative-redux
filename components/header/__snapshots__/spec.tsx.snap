// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component renders the Header component 1`] = `
<View>
  <Text>
    test Title
  </Text>
</View>
`;
